tripy/tripy/frontend/trace/ops/where.py

```diff
@@ -113,26 +115,26 @@ def to_flat_ir(self, inputs, outputs):
                 shape2_name="the previously computed broadcast of the 'input' and 'other' tensors",
             )
 
-            inputs[0] = op_utils.insert_broadcast(
-                inputs[0],
+            broadcasted_input_0 = op_utils.insert_broadcast(
+                broadcasted_input_0,
                 outputs[0].rank,
                 shape_of_target_tensor=computed_output_shape,
                 tensor_details=f"first input of 'where' ('condition')",
             )
-            inputs[1] = op_utils.insert_broadcast(
-                inputs[1],
+            broadcasted_input_1 = op_utils.insert_broadcast(
+                broadcasted_input_1,
                 outputs[0].rank,
                 shape_of_target_tensor=computed_output_shape,
                 tensor_details="second input of 'where' ('input')",
             )
-            inputs[2] = op_utils.insert_broadcast(
-                inputs[2],
+            broadcasted_input_2 = op_utils.insert_broadcast(
+                broadcasted_input_2,
                 outputs[0].rank,
                 shape_of_target_tensor=computed_output_shape,
                 tensor_details="third input of 'where' ('other')",
             )
 
-        SelectOp.build(inputs, outputs)
+        SelectOp.build([broadcasted_input_0, broadcasted_input_1, broadcasted_input_2], outputs)
```

Comment:
Why do you need to make this change? Is it causing issue integrate_subgraph?

Reply:
FlatIR function inputs and outputs accept inputs/outputs which are cloned versions of original tensors such that we have a caller <-> callee relationship among tensors.

Eg. We have a function `Where_1` with those inputs and associate the caller using a `caller_tensor` argument.
```
function Where_1(
    t_inter188: [rank=(0), dtype=(bool), loc=(gpu:0)]
    t_inter189: [rank=(0), dtype=(int32), loc=(gpu:0)]
    t_inter190: [rank=(1), shape=((1,)), dtype=(int32), loc=(gpu:0)]
) -> (
    t_inter191: [rank=(1), dtype=(int32), loc=(gpu:0)]
) {
 ...
}
```

Call site for `Where_1`.
```
t56 = function Where_1(t46, t54, t55)
```

In case you override `inputs` with a function local tensors (in this case "broadcasted tensors"), the relationship to the caller gets overridden.

I have not given a lot of thought to how this can be fixed more efficiently, so please let me know if you have a better approach.

Reply:
Updating `inputs` didn't make sense anyway. This way is better IMO.

Reply:
@parthchadha Resolving this. Let me know if you have any concerns.